learn/what_is_meilisearch/comparison_to_alternatives.md

```diff
@@ -9,12 +9,144 @@
+| Query field weights | ✅ | ✅ | ✅ | ✅ |
+| Synonyms | ✅ | ✅ | ✅ | ✅ |
+| Stop words | ✅ | ✅ | ❌ | ✅ |
+| Automatic language detection | ✅ | ✅ | ❌ | ❌ |
+| All language supports | ✅ | ✅ | ❌ <br> Only space separated | ✅ |
+
+##### Security
+
+| - | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| API Key Management | ✅ | ✅ | ✅ | ✅ |
+| Tenant tokens & multi-tenant indexes | **Q1 2022** <br> Configurable end-user hard filters per index. | ✅  <br> Hard filters are not configurable per index for an end-user tenant key.   | ✅ <br> Hard filters are not configurable per index for an end-user tenant key. | ✅ <br> Role based |
+
+##### Search
+
+| - | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| Placeholder search | ✅ | ✅ | ✅ | ✅ |
+| Multi-index search | Early 2022 | ✅ | ✅ | ✅ |
+| Exact phrase search | ✅ | ✅ | ❌ | ✅ |
+| Geo search |  ✅  | ✅ | ✅ | ✅ |
+| Sort by  |  ✅  | 🔶 <br> Limited to one `sort_by` rule per index. Index have to be duplicated for each sort field and sort order. | 🔶 <br> Do not support sort on string field. | ✅ |
+| Filtering |  ✅ <br>Support complex filter queries with an SQL-like syntax.  | 🔶 <br>Does not support `OR` operation across multiple fields. | 🔶 <br>Does not support `OR` operation across multiple fields. | ✅ |
+| Facetting |  ✅ | ✅ | ✅ | ✅ |
+| Facet value searches | ❌ | ✅ | ✅ | ❌ |
+| Distinct <br><div style="color:#A9A9A9;margin-top:0.8em;">De-duplicate documents by a field value</div>| ✅ | ✅ | ✅  | ✅ |
+| Grouping <br><div style="color:#A9A9A9;margin-top:0.8em;">Bucket documents by field values</div> | ❌ | ✅ | ✅  | ✅ |
+
+##### Vizualize
+
+| - | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| [Visual Dashboard](https://github.com/meilisearch/mini-dashboard) | ✅ | 🔶 <br>Cloud product | 🔶 <br>Cloud product | ✅ |
+
+#### Deploy
+
+| - | MeiliSearch | Algolia | Typesense | Elasticsearch |
+|---|:---:|:----:|:---:|:---:|
+| Self-hosted | ✅  | ❌  | ✅  | ✅ |
+| Official 1-click deploy | ✅ <br> [DigitalOcean](https://marketplace.digitalocean.com/apps/meilisearch) <br> [Platform.sh](https://console.platform.sh/projects/create-project?template=https://raw.githubusercontent.com/platformsh/template-builder/master/templates/meilisearch/.platform.template.yaml) | ❌ | 🔶 <br>Only for the cloud-hosted solution. | ❌ |
+| Official cloud-hosted solution | [Join the beta](https://meilisearch.typeform.com/to/FtnzvZfh?typeform-source=comparative-table) | ✅ | ✅ | ✅ |
+| High availability | Expected 2022 | ✅ | ✅ | ✅ |
+| Run-time dependencies | None | n/a | None | None |
+| Backward compatibility | ✅ | n/a | ✅ | ✅ |
+| Upgrade path | Replace binary, restart process, documents need to be reindexed. | n/a  | Replace binary, restart process, documents need to be reindexed. | Replace binary, restart process, documents need to be reindexed. |
```

Comment:
```suggestion
| Upgrade path | Replace binary, restart process, documents need to be reindexed. | N/A  | Replace binary, restart process, documents need to be reindexed. | Replace binary, restart process, documents need to be reindexed. |
```

Reply:
```suggestion
| Upgrade path | Documents need to be reindexed | n/a  | Documents need to be reindexed | Documents need to be reindexed |
```

Reply:
I would have simplified it in this way to emphasize that there is an incompressible re-indexing time at each upgrade. Also, it's my fault but the initial text is from an external comparison.